flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java

```diff
@@ -151,7 +152,9 @@ public DataStream<RowData> build() {
       RowDataReaderFunction rowDataReaderFunction = new RowDataReaderFunction(
           flinkConf,
           arcticTable.schema(),
-          arcticTable.schema(),
+          projectedSchema == null ?
```

Comment:
Line 144 has checked projectedSchema==null, scanContext.project() can be used here directly.